src/common/types/date.cpp

```diff
@@ -149,6 +154,13 @@ static bool TryConvertDate(const char *buf, date_t &result) {
 		return false;
 	}
 
+	if (strict) {
```

Comment:
Same here, shouldn't this be either !strict or always done?

Reply:
Event in strict casting I want to allow for trailing spaces, just no other character. I made this more explicit. - And fixed a small bug.